esp-hal/src/dma/mod.rs

```diff
@@ -1977,6 +1977,36 @@ pub trait DmaTxBuffer {
     fn length(&self) -> usize;
 }
 
+pub(crate) struct DmaBufferRef<'a, B> {
```

Comment:
I thought it'd be worth pushing [what I'm cooking up](https://github.com/esp-rs/esp-hal/pull/2242) early so we can consider how it will interact with this type.

Reply:
I'm thinking that my approach in this PR is somewhat wrong. I don't need the embedded-hal operations to go through `SpiDmaTransfer` at all, so I think if I cut the functions a bit differently I can avoid about 80% of the current changes.